kafka/conn.py

```diff
@@ -206,8 +218,30 @@ def reinit(self):
         if self._sock:
             self.close()
 
-        try:
-            self._sock = socket.create_connection((self.host, self.port), self.timeout)
-        except socket.error:
-            log.exception('Unable to connect to kafka broker at %s:%d' % (self.host, self.port))
-            self._raise_connection_error()
+        if self.ssl:
+            # Disallow use of SSLv2 and V3 (meaning we require TLSv1.0+)
+            context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+            context.options |= ssl.OP_NO_SSLv2
+            context.options |= ssl.OP_NO_SSLv3
+            context.verify_mode = ssl.CERT_OPTIONAL
+            if self.verify_hostname:
+                context.check_hostname = True
+            if self.ca:
+                context.load_verify_locations(self.ca)
+            if self.certfile and self.keyfile:
+                context.verify_mode = ssl.CERT_REQUIRED
+                context.load_cert_chain(certfile=self.certfile, keyfile=self.keyfile)
+            try:
+                conn = context.wrap_socket(socket.socket(socket.AF_INET),
+                                           server_hostname=self.host)
+                conn.connect((self.host, self.port))
+                self._sock = conn
+            except ssl.SSLError:
+                log.exception('Unable to connect to Kafka broker at %s:%d over SSL', self.host, self.port)
```

Comment:
I'd avoid using `%d` in logging, there's no functional difference to using `%s` or `%r` when someone passes in a number, but it lets just log something useful in case the input was not a valid number (instead of breaking the exception handler)